components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -409,6 +412,9 @@ internal sealed class SubscriptionWrapper : ISubscription {
                         }
                         try {
                             toAdd.Create(rawSubscription.Session, codec, activate);
+                            if (toAdd.Template.EventFilter?.SelectClauses?.Count > 0) {
```

Comment:
Was this possible to do by checking the type instead?

Reply:
Yes, in the next PR were we introduce the base class / derived classes